__tests__/testUtilities/resourceBuilderUtils.js

```diff
@@ -154,7 +154,12 @@ export default class ResourceBuilder {
 
     // Inject property into value
     if (!_.isEmpty(values) && this.injectPropertyIntoValue)
-      values.forEach((value) => (value.propertyKey = property.key))
+      values.forEach((value) => {
+        value.propertyKey = property.key
+        if (value.literal && property?.propertyTemplate?.validationDataType) {
```

Comment:
Not thrilled with this solution. It is kludging in your particular test, but would be unclear in future why this is done. I would prefer to have this be controlled by separate configuration, like `injectPropertyIntoValue`.

From a naming perspective, probably best to rename `injectPropertyIntoValue` to ``injectPropertyKeyIntoValue`, but I defer to your judgment.